src/test/abilities/mycelium_might.test.ts

```diff
@@ -36,71 +36,63 @@ describe("Abilities - Mycelium Might", () => {
     const enemyPokemon = game.scene.getEnemyPokemon();
-    const enemyIndex = enemyPokemon?.getBattlerIndex();
 
     game.doAttack(getMovePosition(game.scene, 0, Moves.BABY_DOLL_EYES));
 
-    await game.phaseInterceptor.to(MovePhase, false);
+    await game.phaseInterceptor.run(EnemyCommandPhase);
+    const phase = game.scene.getCurrentPhase() as TurnStartPhase;
+    const speedOrder = phase.getSpeedOrder();
+    const commandOrder = phase.getCommandOrder();
     // The opponent Pokemon (without Mycelium Might) goes first despite having lower speed than the player Pokemon.
-    expect((game.scene.getCurrentPhase() as MovePhase).pokemon.getBattlerIndex()).toBe(enemyIndex);
-
-    await game.phaseInterceptor.run(MovePhase);
-    await game.phaseInterceptor.to(MovePhase, false);
-
     // The player Pokemon (with Mycelium Might) goes last despite having higher speed than the opponent.
-    expect((game.scene.getCurrentPhase() as MovePhase).pokemon.getBattlerIndex()).toBe(leadIndex);
+    // This means that the commandOrder is equivalent to the speed Order reversed
+    expect(speedOrder.reverse().every((val, index) => val === commandOrder[index])).toBe(true);
     await game.phaseInterceptor.to(TurnEndPhase);
+    // Despite the opponent's ability (Clear Body), its attack stat is still reduced.
     expect(enemyPokemon?.summonData.battleStats[BattleStat.ATK]).toBe(-1);
   }, 20000);
 
   it("Pokemon with Mycelium Might will go first if a status move that is in a higher priority bracket than the opponent's move is used", async() => {
     game.override.enemyMoveset([Moves.TACKLE, Moves.TACKLE, Moves.TACKLE, Moves.TACKLE]);
     await game.startBattle([ Species.REGIELEKI ]);
-
-    const leadIndex = game.scene.getPlayerPokemon()!.getBattlerIndex();
     const enemyPokemon = game.scene.getEnemyPokemon();
-    const enemyIndex = enemyPokemon?.getBattlerIndex();
 
     game.doAttack(getMovePosition(game.scene, 0, Moves.BABY_DOLL_EYES));
 
-    await game.phaseInterceptor.to(MovePhase, false);
+    await game.phaseInterceptor.run(EnemyCommandPhase);
+    const phase = game.scene.getCurrentPhase() as TurnStartPhase;
+    const speedOrder = phase.getSpeedOrder();
+    const commandOrder = phase.getCommandOrder();
     // The player Pokemon (with M.M.) goes first because its move is still within a higher priority bracket than its opponent.
-    expect((game.scene.getCurrentPhase() as MovePhase).pokemon.getBattlerIndex()).toBe(leadIndex);
-
-    await game.phaseInterceptor.run(MovePhase);
-    await game.phaseInterceptor.to(MovePhase, false);
     // The enemy Pokemon goes second because its move is in a lower priority bracket.
-    expect((game.scene.getCurrentPhase() as MovePhase).pokemon.getBattlerIndex()).toBe(enemyIndex);
+    // This means that the commandOrder should be identical to the speedOrder
+    expect(speedOrder.every((val, index) => val === commandOrder[index])).toBe(true);
```

Comment:
```suggestion
    expect(speedOrder).toContainEqual(commandOrder);
```

does this work here? I thought there were matchers for directly comparing two arrays. honestly, I think even `.toBeEqual()` is deep equality by default so maybe that alone works

Reply:
No, it does not work.